frugalos_raft/src/storage/mod.rs

```diff
@@ -94,7 +102,7 @@ impl Storage {
     }
     pub(crate) fn load_log(&mut self, start: LogIndex, end: Option<LogIndex>) -> LoadLog {
         if let Err(e) = track!(self.poll_and_handle_event()) {
-            return LoadLog(log::LoadLogInner::Failed(e));
+            return LoadLog::new(log::LoadLogInner::Failed(e), &self.metrics);
```

Comment:
I suggest as follows.

```
return LoadLog::new(log::LoadLogInner::Failed(e), self.metrics.clone());
```
Others (SaveLog, LoadBallot, SaveBallot, ...) as well.
How is this?

Reply:
fixed